controller/resource_test.go

```diff
@@ -12,127 +31,173 @@ import (
+		task3Name.Partition = partitionValue
+		task3Name.Namespace = namespaces[0]
+		task3.Tags = append(task3.Tags, meshTag, extPartitionTag, namespace1Tag)
+		aclToken1.Partition = partitionValue
+		aclToken1.Namespace = namespaces[0]
+		aclTokenListEntry1.Partition = partitionValue
+		aclTokenListEntry1.Namespace = namespaces[0]
+		aclToken2.Partition = partitionValue
+		aclToken2.Namespace = namespaces[1]
+		aclTokenListEntry2.Partition = partitionValue
+		aclTokenListEntry2.Namespace = namespaces[1]
+		aclToken3.Partition = partitionValue
+		aclToken3.Namespace = namespaces[0]
+		aclTokenListEntry3.Partition = partitionValue
+		aclTokenListEntry3.Namespace = namespaces[0]
+	}
 	cases := map[string]struct {
 		paginateResults bool
 		tasks           []ecs.Task
-		expected        map[string]ServiceState
+		expected        map[ServiceName]ServiceState
 		aclTokens       []*api.ACLToken
 	}{
 		"no overlap between tasks, services and tokens": {
-			tasks:     []ecs.Task{task1, task2},
+			tasks:     []ecs.Task{task1, task2, task3},
 			aclTokens: []*api.ACLToken{aclToken3},
-			expected: map[string]ServiceState{
-				"service1": {
+			expected: map[ServiceName]ServiceState{
+				task1Name: {
 					ConsulECSTasks: true,
 				},
-				"service3": {
+				task3Name: {
 					ACLTokens: []*api.ACLTokenListEntry{aclTokenListEntry3},
 				},
 			},
 		},
 		"all overlap between tasks, services and tokens": {
 			tasks:     []ecs.Task{task1, task2},
-			aclTokens: []*api.ACLToken{aclToken1},
-			expected: map[string]ServiceState{
-				"service1": {
+			aclTokens: []*api.ACLToken{aclToken1, aclToken2},
+			expected: map[ServiceName]ServiceState{
+				task1Name: {
 					ConsulECSTasks: true,
 					ACLTokens:      []*api.ACLTokenListEntry{aclTokenListEntry1},
 				},
+				task2Name: {
+					ConsulECSTasks: true,
+					ACLTokens:      []*api.ACLTokenListEntry{aclTokenListEntry2},
+				},
 			},
 		},
 		"with pagination": {
 			tasks:           []ecs.Task{task1, task2},
 			paginateResults: true,
-			expected: map[string]ServiceState{
-				"service1": {
+			expected: map[ServiceName]ServiceState{
+				task1Name: {
 					ConsulECSTasks: true,
 				},
 			},
 		},
 		"with non-mesh tasks": {
 			tasks:     []ecs.Task{nonMeshTask},
 			aclTokens: []*api.ACLToken{aclToken1},
-			expected: map[string]ServiceState{
-				"service1": {
+			expected: map[ServiceName]ServiceState{
+				task1Name: {
 					ConsulECSTasks: false,
 					ACLTokens:      []*api.ACLTokenListEntry{aclTokenListEntry1},
 				},
 			},
 		},
 	}
 
+	if enterprise {
+		// in the enterprise case the services are qualified by their partition and namespace
+		// and thus become unique entries.. add the missing expected service states.
+		e := cases["no overlap between tasks, services and tokens"].expected
+		e[task2Name] = ServiceState{ConsulECSTasks: true}
+		e = cases["with pagination"].expected
+		e[task2Name] = ServiceState{ConsulECSTasks: true}
```

Comment:
Thinking aloud: Does it make sense to support OSS-specific and Enterprise-specific cases?

So, for example, then we could create separate versions of those cases with the unique oss vs ent setup:

* `"with pagination [enterprise-only]"`
* `"with pagination [oss-only]"`

And then skip oss-only or enterprise-only tests where appropriate?

```go
for name, c := range cases {
    if enterprise && strings.Contains("oss-only", name) {
        continue
    }
    if !enterprise && strings.Contains("enterprise-only", name) {
        continue
    }
}
```

(Or even just have a `c.enterpriseOnly` flag). Thoughts? Not all that important to me, tbh, so can ignore if it doesn't make sense.

Reply:
Generally I think this makes sense, yes. The logic for this particular case is not really about skipping the test entirely, but modifying the behaviour in the enterprise case; specifically, that when ECS tasks with the same name are detected in the cluster belonging to separate namespaces, they become unique services. Which is a case that we want to test here I think.

Admittedly, this test is a bit lacking because I didn't fully re-factor it - I simply made it work in both the OSS and ENT cases.. for example, there is currently no policy validation.. but policy validation is done in the `TestTaskLifecycle` test cases.